src/components/remote_control/src/resource_allocation_manager_impl.cc

```diff
@@ -0,0 +1,371 @@
+    for (; disallowed_modules.end() != module; ++module) {
+      ReleaseResource(*module, application_id);
+      if (rc_extention) {
+        rc_extention->UnsubscribeFromInteriorVehicleData(Json::Value(*module));
+      }
+    }
+  }
+}
+
+RCAppExtensionPtr ResourceAllocationManagerImpl::GetApplicationExtention(
+    application_manager::ApplicationSharedPtr application) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (!application) {
+    return NULL;
+  }
+
+  RCAppExtensionPtr rc_app_extension;
+  application_manager::AppExtensionPtr app_extension =
+      application->QueryInterface(rc_plugin_.GetModuleID());
+  if (!app_extension) {
+    return NULL;
+  }
+
+  rc_app_extension =
+      application_manager::AppExtensionPtr::static_pointer_cast<RCAppExtension>(
+          app_extension);
+
+  return rc_app_extension;
+}
+
+void ResourceAllocationManagerImpl::RemoveAppsSubscriptions(const Apps& apps) {
+  Apps::const_iterator app = apps.begin();
+  for (; apps.end() != app; ++app) {
+    application_manager::ApplicationSharedPtr app_ptr = *app;
+    if (!app_ptr) {
+      continue;
+    }
+    RCAppExtensionPtr rc_extention = GetApplicationExtention(app_ptr);
+    if (rc_extention) {
+      rc_extention->UnsubscribeFromInteriorVehicleData();
+    }
+  }
+}
+
+std::vector<std::string> ResourceAllocationManagerImpl::GetAcquiredResources(
+    const uint32_t application_id) const {
+  LOG4CXX_AUTO_TRACE(logger_);
+  Resources allocated_resources;
+  AllocatedResources::const_iterator allocation = allocated_resources_.begin();
+  for (; allocated_resources_.end() != allocation; ++allocation) {
+    if (application_id == allocation->second) {
+      allocated_resources.push_back(allocation->first);
+    }
+  }
+
+  LOG4CXX_DEBUG(logger_,
+                "Application " << application_id << " acquired "
+                               << allocated_resources.size()
+                               << " resource(s).");
+
+  return allocated_resources;
+}
+
+void ResourceAllocationManagerImpl::SetResourceState(
+    const std::string& module_type,
+    const uint32_t app_id,
+    const ResourceState::eType state) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  LOG4CXX_DEBUG(logger_,
+                "Setting state for " << module_type << " by app_id " << app_id
+                                     << " to state " << state);
+  {
+    sync_primitives::AutoLock lock(allocated_resources_lock_);
+    const AllocatedResources::const_iterator allocated_it =
+        allocated_resources_.find(module_type);
+
+    const std::string status = allocated_resources_.end() != allocated_it
+                                   ? " acquired "
+                                   : " not acquired ";
+    UNUSED(status);
+    LOG4CXX_DEBUG(logger_,
+                  "Resource " << module_type << " is " << status
+                              << " Owner application id is "
+                              << allocated_it->second
```

Comment:
If  the case happens where `allocated_it == allocated_resources_.end()`, then the use of this pointer is invalid here as it is past the end of its container.

Reply:
@JackLivio  issue will be fixed.